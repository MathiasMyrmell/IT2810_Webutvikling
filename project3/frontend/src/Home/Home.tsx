import React from 'react';
import './Home.css';
import { useEffect, useState } from "react";
import RecipeCard from './RecipeCard';
import { useAppSelector, useAppDispatch } from '../Redux/hooks';
import Search from './Search/Search'
import BasicPagination from './BasicPagination';
import { useQuery, gql } from '@apollo/client';

// Type RecipeData
export interface RecipeData {
    id: number,
    title: string,
    directions: Array<string>,
    ingredients: Array<string>
}

// Query with search, filter and sorting
function getRecipesQuery(recipesPerPage: number, page: number, sort: string, where: string, selected: string[]) {
    if (sort !== "DESC" && sort !== "ASC") { sort = "DESC"; }
    const offset = (page - 1) * recipesPerPage;

    const includedIngredients = selected.length ? "AND: [" + selected.map(s => '{ingredients_INCLUDES: "' + s + '"}').join(", ") + "]" : "";

    return gql`
    {
        recipes(options: { offset: ${offset}, limit: ${recipesPerPage}, sort: { title: ${sort} } }, where: {title_CONTAINS: "${where}"${includedIngredients}}) {
            title
            directions
            ingredients
          }
    }`;
}

// To get number of titles
function getTitlesQuery(where: string, selected: string[]) {
    const includedIngredients = selected.length ? "AND: [" + selected.map(s => '{ingredients_INCLUDES: "' + s + '"}').join(", ") + "]" : "";

    return gql`
    {
        recipes(where: {title_CONTAINS: "${where}"${includedIngredients}}) {
            title
          }
    }`;
}
// Length of all objects for pagination
function useTitleCount(where: string, selected: string[]) {
    const { loading, error, data } = useQuery(getTitlesQuery(where, selected));
    return (data ? data.recipes.length : 10)
}

interface dataResult {
    recipes: Array<RecipeData>
}
interface mockProp {
    mockData?: dataResult;
}

let oldResult: dataResult;
let oldResultCount: number = 1;
function Home(props: dataResult) {

    const recipesPerPage: number = 10;
    const selected = useAppSelector(state => state.sort);
    const { loading, error, data } = useQuery(getRecipesQuery(recipesPerPage, selected.currentPage, selected.sorted, selected.title, selected.selectedIngredients));
    const recipeCount = useTitleCount(selected.title, selected.selectedIngredients);

    // If we get error
    if (error) return <h1>Error: {error.message}</h1>

    // Inserting mock data into oldResult
    if (props.recipes.length) { oldResult = props }

    // Save last fetched data
    if (data) {
        oldResult = data;
        oldResultCount = recipeCount;
    }
    return (
        <div role="main" className="Home">
            <h1>Recipes</h1>
            <div className='flexMain'>
                <Search />
                <div className='recipes'>
                    <BasicPagination recipesPerPage={recipesPerPage} totalRecipe={loading ? oldResultCount : recipeCount} />
                    {/* Render list of recipes if fetched, previous results if not. If no previous results are stored return empty element. */}
                    {(!oldResult && loading) ? <></> : ((data ? data : oldResult).recipes).map((d: RecipeData, i: number) => <RecipeCard key={i} {...d} />)}
                </div>
            </div>
        </div>
    );
}

export default Home;
