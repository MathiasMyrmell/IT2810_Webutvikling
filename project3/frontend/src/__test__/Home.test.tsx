import { Provider } from 'react-redux';
import { ApolloProvider, ApolloClient, InMemoryCache } from '@apollo/client'
import { render } from '@testing-library/react';
import store from '../Redux/store';
import renderer from 'react-test-renderer';
import Home from '../Home/Home';
import { RecipeData } from '../Home/Home';
import React from 'react';

const client = new ApolloClient({
    uri: 'http://it2810-30.idi.ntnu.no:8080/graphql',
    cache: new InMemoryCache(),
});

interface dataResult {
    recipes: Array<RecipeData>
}

const mock: dataResult = {
    recipes: [
        { id: 0, title: "test0", directions: ["a", "b"], ingredients: ["c, d"] },
        { id: 1, title: "test1", directions: ["e", "f"], ingredients: ["f, g"] }
    ]
}

test('Snapshot test Home with mock data', () => {
    const tree = renderer.create(
        <ApolloProvider client={client}>
            <React.StrictMode>
                <Provider store={store}>
                    <Home {...mock} />
                </Provider>
            </React.StrictMode>
        </ApolloProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});