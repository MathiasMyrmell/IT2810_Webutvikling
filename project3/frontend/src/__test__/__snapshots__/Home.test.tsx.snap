// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Home with mock data 1`] = `
<div
  className="Home"
  role="main"
>
  <h1>
    Recipes
  </h1>
  <div
    className="flexMain"
  >
    <div
      className="Search"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1m9tqjn-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h2>
            Search
          </h2>
          <div
            className="MuiFormControl-root MuiTextField-root css-1lwde3r-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="searchField"
              id="searchField-label"
            >
              Title
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="searchField"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Title
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <h2>
            Select
          </h2>
          <div
            className="SearchForm"
          >
            <div
              className="MuiBox-root css-k008qs"
            >
              <fieldset
                className="MuiFormControl-root css-1fc7482-MuiFormControl-root"
              >
                <div
                  className="MuiFormGroup-root MuiFormGroup-row css-qfz70r-MuiFormGroup-root"
                >
                  <label
                    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      onBlur={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate={false}
                        disabled={false}
                        name="chicken"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      chicken
                    </span>
                  </label>
                  <label
                    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      onBlur={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate={false}
                        disabled={false}
                        name="beef"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      beef
                    </span>
                  </label>
                  <label
                    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      onBlur={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate={false}
                        disabled={false}
                        name="milk"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      milk
                    </span>
                  </label>
                  <label
                    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      onBlur={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate={false}
                        disabled={false}
                        name="butter"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      butter
                    </span>
                  </label>
                  <label
                    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      onBlur={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate={false}
                        disabled={false}
                        name="cheese"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      cheese
                    </span>
                  </label>
                  <label
                    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      onBlur={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate={false}
                        disabled={false}
                        name="onion"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      onion
                    </span>
                  </label>
                  <label
                    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      onBlur={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate={false}
                        disabled={false}
                        name="broccoli"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      broccoli
                    </span>
                  </label>
                  <label
                    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      onBlur={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate={false}
                        disabled={false}
                        name="flour"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      flour
                    </span>
                  </label>
                  <label
                    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      onBlur={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate={false}
                        disabled={false}
                        name="tomato"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      tomato
                    </span>
                  </label>
                  <label
                    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      onBlur={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate={false}
                        disabled={false}
                        name="garlic"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      garlic
                    </span>
                  </label>
                </div>
              </fieldset>
            </div>
          </div>
          <h2>
            Sort
          </h2>
          <select
            aria-label="for sorting"
            className="form-control"
            defaultValue="A-Z"
            onChange={[Function]}
          >
            <option
              value="A-Z"
            >
              A-Z
            </option>
            <option
              value="Z-A"
            >
              Z-A
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="recipes"
    >
      <div
        className="BasicPagination"
      >
        <div
          className="css-1p5q5e5-MuiStack-root"
        >
          <nav
            aria-label="pagination navigation"
            className="MuiPagination-root MuiPagination-outlined css-1oj2twp-MuiPagination-root"
          >
            <ul
              className="MuiPagination-ul makeStyles-ul-1 css-wjh20t-MuiPagination-ul"
            >
              <li>
                <button
                  aria-label="Go to previous page"
                  className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeLarge MuiPaginationItem-outlined MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-ax94ij-MuiButtonBase-root-MuiPaginationItem-root"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-n8417t-MuiSvgIcon-root-MuiPaginationItem-icon"
                    data-testid="NavigateBeforeIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                </button>
              </li>
              <li>
                <button
                  aria-current="true"
                  aria-label="page 1"
                  className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeLarge MuiPaginationItem-outlined MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-ax94ij-MuiButtonBase-root-MuiPaginationItem-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  1
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to next page"
                  className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeLarge MuiPaginationItem-outlined MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-ax94ij-MuiButtonBase-root-MuiPaginationItem-root"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-n8417t-MuiSvgIcon-root-MuiPaginationItem-icon"
                    data-testid="NavigateNextIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1muwu1d-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiTypography-h4 MuiCardHeader-title css-9o3s4j-MuiTypography-root"
              >
                test0
              </span>
            </div>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiCardContent-root css-h8hf1i-MuiCardContent-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-z2eky3-MuiTypography-root"
            >
              Click for more information
            </p>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
        </button>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1muwu1d-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiTypography-h4 MuiCardHeader-title css-9o3s4j-MuiTypography-root"
              >
                test1
              </span>
            </div>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiCardContent-root css-h8hf1i-MuiCardContent-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-z2eky3-MuiTypography-root"
            >
              Click for more information
            </p>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
